### PR TITLE
Allow ignoring function ref issues

### PR DESCRIPTION
**Pull request by Sushisource**
Allows `//workflowcheck:ignore` to work on workflow function registration sites

### CODE REVIEW

**Review by Quinn-With-Two-Ns — APPROVED** [review]

File: contrib/tools/workflowcheck/workflow/checker.go [hunk 1]
```diff
@@ -176,7 +180,17 @@ func (c *Checker) Run(pass *analysis.Pass) error {
 			}
 			// Report if couldn't get type
 			if fn == nil {
-				pass.Reportf(callExpr.Args[0].Pos(), "unrecognized function reference format")
+				_, isIgnored := ignoreMap[n]
```
Comment (cretz) [inline contrib/tools/workflowcheck/workflow/checker.go:183]: I think a better impl is to change `if callExpr == nil {` above to `if callExpr == nil || isIgnored {`. You are ignoring `RegisterWorkflow` call so check it early.

File: contrib/tools/workflowcheck/workflow/checker.go [hunk 2]
```diff
@@ -152,10 +152,21 @@ func (c *Checker) Run(pass *analysis.Pass) error {
...
 		ast.Inspect(file, func(n ast.Node) bool {
-			// Only handle calls
+			// Only handle calls with followable function pointers
+			_, isIgnored := ignoreMap[n]
+			for k := range ignoreMap {
+				asExprStmt, _ := k.(*ast.ExprStmt)
+				if asExprStmt != nil && asExprStmt.X == n {
+					isIgnored = true
+				}
+			}
```
Comment (cretz) [inline contrib/tools/workflowcheck/workflow/checker.go:167]: Is this for loop needed?

Comment (cretz) [inline contrib/tools/workflowcheck/workflow/checker.go:167]: In retrospect, you could have just:

```suggestion
			if _, isIgnored := ignoreMap[n]; isIgnored {
				return false
			}
```

Comment (Sushisource) [reply contrib/tools/workflowcheck/workflow/checker.go:167]: That didn't seem to work when I was trying it - I needed to get this inner node for things to work

**Pull request merged**